core/blockchain.go

```diff
@@ -739,19 +739,19 @@ func (bc *BlockChain) Export(w io.Writer) error {
 
 // ExportN writes a subset of the active chain to the given writer.
 func (bc *BlockChain) ExportN(w io.Writer, first uint64, last uint64) error {
-	if !bc.chainmu.TryLock() {
-		return errChainStopped
-	}
-	defer bc.chainmu.Unlock()
-
 	if first > last {
 		return fmt.Errorf("export failed: first (%d) is greater than last (%d)", first, last)
 	}
 	log.Info("Exporting batch of blocks", "count", last-first+1)
 
 	start, reported := time.Now(), time.Now()
+	var parentHash common.Hash
 	for nr := first; nr <= last; nr++ {
 		block := bc.GetBlockByNumber(nr)
+		if nr > first && block.Hash() != parentHash {
```

Comment:
You'll need to do this check after ensuring that the block is not nil. Otherwise .Hash will panic.

Reply:
🤦 done